include/matx/generators/random.h

```diff
@@ -186,233 +206,62 @@ template <typename T> class [[deprecated("Use random() operator instead of rando
-    }
-
-    return alpha_ * val + beta_;
+  template< typename T >
+  struct randIntParams{
+    T min_;
+    T max_;
   };
-
-  /**
-   * Get rank of random view
-   *
-   * @return Rank of view
-   */
-  static inline constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank() { return RANK; }
-
-  /**
-   *  Get size of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Size of dimension
-   */
-  constexpr index_t inline __MATX_HOST__ __MATX_DEVICE__ Size(int dim) const
-  {
-    return shape_[dim];
-  }
-
-  /**
-   * Get stride of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Stride of dimension
-   */
-  index_t inline __MATX_HOST__ __MATX_DEVICE__ Stride(int dim) const
-  {
-    return strides_[dim];
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PreRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept
-  {
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PostRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept  
-  {
-  }  
-};
-
-  namespace detail {
+  
+  template< typename T >
+  struct randFloatParams{
+    Distribution_t dist_;
+    T alpha_;
+    T beta_;
+  };  
+  
     template <typename T, typename ShapeType>
     class RandomOp : public BaseOp<RandomOp<T, ShapeType>> {
       private:
         using inner_t = typename inner_op_type_t<T>::type;
         static constexpr int RANK = cuda::std::tuple_size<ShapeType>{};
-        Distribution_t dist_;
         cuda::std::array<index_t, RANK> shape_;
         cuda::std::array<index_t, RANK> strides_;
         index_t total_size_;
         curandStatePhilox4_32_10_t *states_;
-        uint64_t seed_;
-        inner_t alpha_;
-        inner_t beta_;
+        uint64_t seed_;     
         bool init_ = false;
         bool device_;
-
+        
+        union{
+        randFloatParams<inner_t> fParams_;
+        randIntParams<inner_t> iParams_;
+        };
+        
         // Used by host operators only
         curandGenerator_t gen_;
         //T val;
 
 
       public:
         using value_type = T;
-        using matxop = bool;
-
-        static_assert(std::is_same_v<T, float> || 
-                      std::is_same_v<T, double> ||
-                      std::is_same_v<T, cuda::std::complex<float>> ||
-                      std::is_same_v<T, cuda::std::complex<double>>);        
+        using matxop = bool;     
 
         __MATX_INLINE__ std::string str() const { return "random"; }
 
         // Shapeless constructor to be allocated at run invocation
         RandomOp() = delete;
-
-        inline RandomOp(ShapeType &&s, Distribution_t dist, uint64_t seed, inner_t alpha, inner_t beta) :
-          dist_(dist), seed_(seed), alpha_(alpha), beta_(beta)
+  
+        // base constructor, should never be called directly
+        inline RandomOp(ShapeType &&s, uint64_t seed) :
```

Comment:
Can you change these to `__MATX_INLINE__` instead to be consistent with other files? I know it was like that already.

Reply:
yep, updated all in this file.